src/logger.rs

```diff
@@ -0,0 +1,47 @@
+//! Module containing logger implementation.
+
+use crate::ffi::string::{AscStr, AscString};
+use log::{Level, LevelFilter, Log, Metadata, Record};
+
+/// The main logger implementation for the `log` facade crate.
+pub struct Logger;
+
+impl Log for Logger {
+    fn enabled(&self, _: &Metadata) -> bool {
+        true
+    }
+
+    fn log(&self, record: &Record) {
+        let level = match record.level() {
+            Level::Error => ERROR,
+            Level::Warn => WARNING,
+            Level::Info => INFO,
+            Level::Debug | Level::Trace => DEBUG,
+        };
+        let message = AscString::new(record.args().to_string());
```

Comment:
There is something like [CString](https://doc.rust-lang.org/std/ffi/struct.CString.html), why not use that instead, if you are linking this to a C lib?